go/tasks/plugins/k8s/sagemaker/sagemaker.go

```diff
@@ -78,10 +78,16 @@ func (m awsSagemakerPlugin) BuildResourceForTrainingJob(
 	if !ok {
 		return nil, errors.Errorf("Required input not specified: [train]")
 	}
+	if trainPathLiteral.GetScalar() == nil || trainPathLiteral.GetScalar().GetBlob() == nil {
+		return nil, errors.Errorf("[train] Input is required and should be of Type [Scalar.Blob]")
+	}
 	validatePathLiteral, ok := inputLiterals["validation"]
 	if !ok {
 		return nil, errors.Errorf("Required input not specified: [validation]")
 	}
+	if validatePathLiteral.GetScalar() == nil || validatePathLiteral.GetScalar().GetBlob() == nil {
```

Comment:
@bnsblue can we find more cases in which we have a potential NPE and fix them before we merge?

Reply:
sure

Reply:
If you haven't, please add unit tests to these functions to cover branches... aim for 80%+ code coverage if we aren't there yet.. for this package at least

Reply:
ok then we should probably merge https://github.com/lyft/flytepropeller-private/pull/257/files alone first